spec.html

```diff
@@ -12349,33 +12349,33 @@ <h1>Objectives</h1>
+        WeakRef-obliviousness, together with liveness, capture two notions. One, that a WeakRef itself does not keep a referent alive. Two, that cycles in liveness does not imply that a referent is live. To be concrete, if determining _ref_'s liveness depends on determining the liveness of another WeakRef referent, _ref2_, _ref2_'s liveness cannot assume _ref_'s liveness, which would be circular reasoning.
       </emu-note>
       <emu-note>
-        WeakRef-obliviousness is defined on sets of objects instead of individual objects to account for cycles. If it were defined on individual objects, then an object in a cycle will be considered live even though its Object value is only observed via WeakRefs of other objects in the cycle.
+        WeakRef-obliviousness is defined on sets of symbols or objects instead of individual referents to account for cycles. If it were defined on individual values, then a referent in a cycle will be considered live even though its respective value is only observed via WeakRefs of other referents in the cycle.
       </emu-note>
       <emu-note>
-        Colloquially, we say that an individual object is live if every set of objects containing it is live.
+        Colloquially, we say that an individual symbol or object is live if every set of symbols and/or objects containing it is live.
       </emu-note>
 
-      <p>At any point during evaluation, a set of objects _S_ is considered <dfn>live</dfn> if either of the following conditions is met:</p>
+      <p>At any point during evaluation, a set of symbols and/or objects _S_ is considered <dfn>live</dfn> if either of the following conditions is met:</p>
 
       <ul>
         <li>
           Any element in _S_ is included in any agent's [[KeptAlive]] List.
         </li>
         <li>
-          There exists a valid future hypothetical WeakRef-oblivious execution with respect to _S_ that observes the Object value of any object in _S_.
+          There exists a valid future hypothetical WeakRef-oblivious execution with respect to _S_ that observes any value in _S_.
         </li>
       </ul>
       <emu-note>
-        The second condition above intends to capture the intuition that an object is live if its identity is observable via non-WeakRef means. An object's identity may be observed by observing a strict equality comparison between objects or observing the object being used as key in a Map.
+        The second condition above intends to capture the intuition that a referent is live if its identity is observable via non-WeakRef means. A referent's identity may be observed by observing a strict equality comparison or observing the value being used as key in a Map.
```

Comment:
I'm more than happy to have this change, though curious what made you think to add it? In my mind's pseudo-mark-sweep-gc of the spec, there is always at least one live realm during execution and the AOs and intrinsic functions are like a _static roots_, so because they reference the well-known symbols then the symbols are reachable from the GC - even if not reachable from userland.

@rricard and I did briefly talk about if we should add a concrete addition to the definition of liveness that says well-known symbols are always considered live. What do you think?

Reply:
PR updated to make the liveness of well-known symbols explicit.

Reply:
Can we move or duplicate the editor's note about well-known symbol under this section.

It's not immediately clear why well-known symbols are not covered by the second bullet point. While some well-known symbols are syntactically undeniable, technically an execution where the deniable well-known symbols were removed from the `Symbol` global and where no new realm could be created wouldn't allow their observation.

Reply:
FWIW well-known symbols can't be removed from the global `Symbol`, they are non-writable and non-configurable. To deny access the whole `Symbol` object needs to be denied and any other operation that could expose the symbol.

```js
// first run before all other code:
delete Symbol.prototype.constructor;
delete gbloalThis.Symbol;
delete Array.prototype.concat;
delete globalThis.ShadowRealm; // and all other ways to construct a new realm
// No longer an execution that could observe the identity of `Symbol.isConcatSpreadable`
```

There is the question of how much flexibility to give engines here. In the above situation should they be allowed to GC `@@isConcatSpreadable`. The idea behind the 3rd bullet point (saying that well-known symbols are always considered live) is to deny that ability to ensure consistency across engines and avoid ambiguity in the spec. Though I can appreciate if engines would rather have the freedom.

cc: @syg @codehag

Reply:
Thanks @littledan , that's a good point. If we start to specifically mention well-known symbol here it starts to raise the question why other thing are/are-not also specified.

I've backed out the normative change. I'll re-add later as an editors note.